bindings/experimental/distrdf/python/DistRDF/HeadNode.py

```diff
@@ -53,7 +82,43 @@ def __nonzero__(self):
         return self.__bool__()
 
 
-class HeadNode(Node.Node):
+class EmptySourceHeadNode(Node.Node):
+    """
+    The head node of a computation graph where the RDataFrame is created from
+    a number of entries.
+    """
+
+    def __init__(self, nentries):
+        """
+        Creates a new RDataFrame instance for the given arguments.
+
+        Args:
+            nentries (int): The number of entries this RDataFrame will process.
+        """
+        super(EmptySourceHeadNode, self).__init__(None, None)
+
+        self.nentries = nentries
+        # Set at creation of the dataframe, might be optimized by the backend
+        # in optimize_partitions
+        self.npartitions = 2
+
+    def build_ranges(self):
+        """Build the ranges for this dataset."""
+        # Empty datasets cannot be processed distributedly
+        if not self.nentries:
+            raise RuntimeError(
+                ("Cannot build a distributed RDataFrame with zero entries. "
+                 "Distributed computation will fail. "))
+        if self.npartitions > self.nentries:
+            # Restrict 'npartitions' if it's greater than 'nentries'
+            msg = ("Number of partitions {0} is greater than number of entries {1} "
+                   "in the dataframe. Using {1} partition(s)".format(self.npartitions, self.nentries))
+            warnings.warn(msg, UserWarning, stacklevel=2)
```

Comment:
(you mentioned [here](https://github.com/root-project/root/pull/8391#discussion_r654544704) that distRDF is also not ok with `nentries == 0`: that case should probably also just work (just run locally and produce a bunch of empty histograms, _maybe_ with a warning) but i'm also ok with leaving that behavior as it is. on the other hand, warning on 1 entry but being ok with 3 is weird)

Reply:
This depends on whether we can always guarantee that backends will be able to process the case of `npartitions == 1`. I think it can be done in general, not sure that is for this PR